src/AnnotationDialog.js

```diff
@@ -565,34 +559,79 @@ class AnnotationDialog extends EventEmitter {
         const annotationEl = document.createElement('div');
         annotationEl.classList.add(CLASS_COMMENT);
         annotationEl.setAttribute('data-annotation-id', annotation.annotationID);
-        annotationEl.innerHTML = `
-            <div class="profile-image-container">${avatarHtml}</div>
-            <div class="profile-container">
-                <div class="user-name">${userName}</div>
-                <div class="comment-date">${created}</div>
-            </div>
-            <div class="comment-text">${text}</div>
-            <button class="bp-btn-plain ${CLASS_BUTTON_DELETE_COMMENT} ${annotation.permissions.can_delete
-    ? ''
-    : constants.CLASS_HIDDEN}" data-type="${constants.DATA_TYPE_DELETE}" title="${this.localized.deleteButton}">
-                ${ICON_DELETE}
-            </button>
-            <div class="${CLASS_DELETE_CONFIRMATION} ${constants.CLASS_HIDDEN}">
-                <div class="delete-confirmation-message">
-                    ${this.localized.deleteConfirmation}
-                </div>
-                <div class="${constants.CLASS_BUTTON_CONTAINER}">
-                    <button class="bp-btn ${CLASS_CANCEL_DELETE}" data-type="${constants.DATA_TYPE_CANCEL_DELETE}">
-                        ${this.localized.cancelButton}
-                    </button>
-                    <button class="bp-btn bp-btn-primary ${CLASS_BUTTON_DELETE_CONFIRM}" data-type="${constants.DATA_TYPE_CONFIRM_DELETE}">
-                        ${this.localized.deleteButton}
-                    </button>
-                </div>
-            </div>`.trim();
 
         const annotationContainerEl = this.dialogEl.querySelector(`.${CLASS_COMMENTS_CONTAINER}`);
         annotationContainerEl.appendChild(annotationEl);
+
+        // Avatar
+        const avatarEl = document.createElement('div');
```

Comment:
I hate to be that guy, but is it possible to break this down into smaller functions, or something, so that it isn't such a big function?